lua/ttt2/libraries/events.lua

```diff
@@ -358,32 +279,38 @@ else --CLIENT
 	end)
 end
 
--- load the events itself
-fileloader.LoadFolder(
+local function ShouldInherit(t, base)
+	return t.base ~= t.type
+end
+
+eventTypes = classbuilder.BuildFromFolder(
 	"terrortown/gamemode/shared/events/",
-	false, -- deepsearch
 	SHARED_FILE,
-	function(path)
-		events.Initialize(path)
+	"EVENT", -- class scope
+	function(class, path, name) -- on class loaded
+		class.type = name
+		class.base = class.base or "base_event"
+
+		_G["EVENT_" .. string.upper(name)] = name
 
 		MsgN("Added TTT2 event file: ", path)
 	end,
-	function() -- pre folder loaded
-		events.InitVariables()
-	end
+	true, -- should inherit
+	ShouldInherit -- special inheritance check
 )
 
-fileloader.LoadFolder(
+table.Merge(eventTypes, classbuilder.BuildFromFolder(
 	"terrortown/events/",
-	false,
 	SHARED_FILE,
-	function(path)
-		events.Initialize(path)
+	"EVENT", -- class scope
+	function(class, path, name) -- on class loaded
```

Comment:
Can this function be localized and then referenced twice? Instead of two functions with the same content.

Reply:
https://github.com/TTT-2/TTT2/pull/692#discussion_r520749256

Reply:
small reminder to check out the linked comment and possibly update your review :)

Reply:
I did see that, but my comment still holds.